mypyc/primitives/str_ops.py

```diff
@@ -134,3 +135,10 @@
     c_function_name="CPyStr_IsTrue",
     error_kind=ERR_NEVER,
 )
+
+str_ssize_t_size_op = custom_op(
+    arg_types=[str_rprimitive],
+    return_type=c_pyssize_t_rprimitive,
+    c_function_name='PyUnicode_GET_LENGTH',
```

Comment:
not a full review, but I think using `PyUnicode_GET_LENGTH` here breaks the design since it's a macro not a function.

https://github.com/python/cpython/blob/9ad8f109ac037ac088e09dcd2da523322c5caf34/Include/cpython/unicodeobject.h#L412

Reply:
we use PyUnicode_GET_LENGTH in many places already and it's a public API according to https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_GET_LENGTH

Reply:
The problem is not about whether it is a public API or not, the problem is about it being a macro. As you already saw when defining the custom_op, the field is called `c_function_name`, we want the called object to be a function instead of a macro which will be textually substituted after preprocessing. Macro also heavily limits the compiler's expressiveness when we are porting to backends with no macro support.

Reply:
Thanks for pointing out. Maybe I should write a wrapper function for PyUnicode_READY() and PyUnicode_GET_LENGTH()?

Reply:
I took a look at their cpython implementation. It's a bit of tricky to implement them directly using mypyc IR. So I guess wrapper is the way to go, at least for now

Reply:
I am going to keep calling `PyUnicode_GET_LENGTH` directly at least for this PR, since `PyUnicode_READY` would be deprecated in 3.12.